YAKL_alloc_free.h

```diff
@@ -75,26 +75,28 @@ namespace yakl {
         alloc = [] ( size_t bytes ) -> void* {
           if (bytes == 0) return nullptr;
           void *ptr = sycl::malloc_shared(bytes,sycl_default_stream);
+          sycl_default_stream.memset(ptr, 0, bytes).wait();
```

Comment:
Mind if I remove this, or would SYCL throw an error if I remove it? The other backends don't initialize to zero for the allocation.

Reply:
The SYCL backend was giving incorrect results from reductions. In short, reduction variables are reusing the values from previous runs. But this `memset` is a temporary patch. Will removed this and replace with a permanent solution.